test/common.jl

```diff
@@ -9,19 +9,48 @@ f(u,p,t) = u
     sol = solve(prob, TaylorMethod(50), abstol=1e-20)
 
     @test sol[end] - 0.5*exp(1) < 1e-12
-    
-    dt = 0.01
-    t = collect(tspan[1]:dt:tspan[2])
-    sol_dt = solve(prob, TaylorMethod(50), abstol=1e-20, saveat = dt)
-    @test all(sol_dt.t .== t)
 end
 
-f(du, u, p, t) = (du .= u)
+f!(du, u, p, t) = (du .= u)
 @testset "Test integration of ODE with abstract arrays in common interface" begin
     u0 = rand(4, 2)
     tspan = (0.0, 1.0)
-    prob = ODEProblem(f, u0, tspan)
+    prob = ODEProblem(f!, u0, tspan)
     sol = solve(prob, TaylorMethod(50), abstol=1e-20)
 
     @test norm(sol[end] - u0.*exp(1)) < 1e-12
 end
+
+u0 = 1.0
+tspan = (0.0,5.0)
+saveat_inputs = ([], 0:1:tspan[2], 0:1:(tspan[2]+5), 3:1:tspan[2], collect(0:1:tspan[2]))
+@testset "Test saveat behavior with numbers in common interface" begin
```

Comment:
`saveat_input` lists various use cases: empty, range within tspan, range outside of tspan, range starting after the start time and a list

Reply:
Great! Thanks for implementing the various cases and adding the corresponding test cases!